waltid-services/waltid-issuer-api/src/test/kotlin/id/walt/LspPotentialTest.kt

```diff
@@ -0,0 +1,86 @@
+    val cwt = Cbor.decodeFromByteArray(COSESign1.serializer(), cwtStr.base64UrlDecode())
+    assertNotNull(cwt.payload)
+    val cwtPayload = Cbor.decodeFromByteArray<MapElement>(cwt.payload!!)
+    assertEquals(DEType.textString, cwtPayload.value.get(MapKey(ProofOfPossession.CWTProofBuilder.LABEL_ISS))?.type)
+    val cwtProtectedHeader = Cbor.decodeFromByteArray<MapElement>(cwt.protectedHeader)
+    assertTrue(cwtProtectedHeader.value[MapKey(ProofOfPossession.CWTProofBuilder.HEADER_LABEL_ALG)]!!.internalValue!! == -7L)
+    assertTrue(cwtProtectedHeader.value[MapKey(ProofOfPossession.CWTProofBuilder.HEADER_LABEL_CONTENT_TYPE)]!!.internalValue!!.equals("openid4vci-proof+cwt"))
+
+    val tokenHeader = cwt.decodeProtectedHeader()
+    val rawKey = (tokenHeader.value[MapKey(ProofOfPossession.CWTProofBuilder.HEADER_LABEL_COSE_KEY)] as ByteStringElement).value
+    val cryptoProvider = SimpleCOSECryptoProvider(listOf(COSECryptoProviderKeyInfo("pub-key", AlgorithmID.ECDSA_256,
+      OneKey(CBORObject.DecodeFromBytes(rawKey)).AsPublicKey()
+    )))
+    //assertTrue(cryptoProvider.verify1(cwt, "pub-key"))
+
+  }
+
+  //@Test
+  fun testASitCredentialRequest() {
+    val reqBody = "{\"format\":\"mso_mdoc\",\"credential_identifier\":null,\"credential_response_encryption\":null,\"doctype\":\"org.iso.18013.5.1.mDL\",\"claims\":null,\"credential_definition\":null,\"vct\":null,\"proof\":{\"proof_type\":\"cwt\",\"jwt\":null,\"cwt\":\"hFkBdaQBJgN0b3BlbmlkNHZjaS1wcm9vZitjd3QEWDlkaWQ6a2V5OnpEbmFldVpxenBudnFTMThiOTVXbXdMdmlrTWZyUjJrRGpSWUVFV0xvQUU4eWVicm8YIVkBGzCCARcwgb6gAwIBAgIIL5E2MUilrikwCgYIKoZIzj0EAwIwEjEQMA4GA1UEAwwHRGVmYXVsdDAeFw0yNDA2MjQxMDA0MDdaFw0yNDA2MjQxMDA0MzdaMBIxEDAOBgNVBAMMB0RlZmF1bHQwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAASw55kB7ZDt6FLVLEiJrQoMutGOSL6wintZ28Q6vPX8eO-o6-iH2qXBBO9h_4JlALgh0q8cPXczGR-GFPhWQ9ALMAoGCCqGSM49BAMCA0gAMEUCIQC8foX3SsFomX14ZLffSKalmABj4Mdr13KG0xd33NfrYwIgFFYZEeyFCYJpwlFDi0XCvmRoYEwqznJTBmkifoDGyQmgWHekAXgbaHR0cHM6Ly93YWxsZXQuYS1zaXQuYXQvYXBwA3goaHR0cHM6Ly9pc3N1ZXIucG90ZW50aWFsLndhbHQtdGVzdC5jbG91ZAYaZnlEmApYJDIyMmUzYTZlLWM2ZTctNGQ0Ni1hMjhlLWQ4ODIyN2JlZmIzYVhA58GZhhZPvdefAJ46vvoUVDSs-0y41E_J4m95QbsgDDPHjbuD3778mnhJwB3HSCKkK96L8jrgsQSgFpqeIGqKTw\"}}"
+    val req = CredentialRequest.fromJSONString(reqBody)
+    assertEquals(CredentialFormat.mso_mdoc, req.format)
+    val coseSign1 = Cbor.decodeFromByteArray<COSESign1>(req.proof!!.cwt!!.base64UrlDecode())
+    assertNotNull(coseSign1.payload)
+    val tokenHeader = coseSign1.decodeProtectedHeader()
+    val x5c = X509CertUtils.parse((tokenHeader.value[MapKey(ProofOfPossession.CWTProofBuilder.HEADER_LABEL_X5CHAIN)] as ByteStringElement).value)
+    val keyId = (tokenHeader.value[MapKey(4)] as ByteStringElement).value.decodeToString()
+    //val rawKey = (tokenHeader.value[MapKey(ProofOfPossession.CWTProofBuilder.HEADER_LABEL_COSE_KEY)] as ByteStringElement).value
```

Comment:
could this be cleaned up? or maybe it's still relevant?

Reply:
that's been cleaned up in the follow-up branch